geopandas/tests/test_geoseries.py

```diff
@@ -152,6 +152,17 @@ def test_coord_slice(self):
         self.assertTrue(geom_equals(self.g3[[True, False]], self.g3.cx[0.9:, :0.1]))
         self.assertTrue(geom_equals(self.g3[[False, True]], self.g3.cx[0:0.1, 0.9:1.0]))
 
+    def test_coord_slice_with_zero(self):
+        """Test that CoordinateSlice correctly handles zero slice (#GH477)."""
```

Comment:
Thanks for the tip! I wasn't familiar with that. There're a handful of those instances elsewhere, so maybe a small item to clean up later.